tutorials/parquet-catalog-demos/euclid-hats-parquet.md

```diff
@@ -0,0 +1,1130 @@
+There are up to 5 redshift estimates per 'OBJECT_ID', per HDU.
+For the Parquet, these were pivoted so that there is one row per 'OBJECT_ID' in order to facilitate the table joins.
+The resulting columns were named by combining the table name (Z), the HDU name, the original column name, and the rank of the given redshift estimate (i.e., the value in the original 'SPE_RANK' column).
+For example, the 'SPE_PDF' column for the highest ranked redshift estimate in the 'GALAXY_CANDIDATES' table is called 'Z_GALAXY_CANDIDATES_SPE_PDF_RANK0'.
+
+Here, we follow IRSA's
+[Cloud Access notebook](https://caltech-ipac.github.io/irsa-tutorials/tutorials/cloud_access/cloud-access-intro.html#navigate-a-catalog-and-perform-a-basic-query)
+to inspect the parquet schema.
+
+```{code-cell}
+s3_filesystem = pyarrow.fs.S3FileSystem()
+schema = pyarrow.parquet.read_schema(euclid_parquet_schema_path, filesystem=s3_filesystem)
+```
+
+There are more than 1300 columns in this dataset.
+
+```{code-cell}
+print(f"{len(schema)} columns total")
+```
+
+### A.1 Euclid Q1 columns
+
++++
+
+To find all columns from a given table, search for column names that start with the table name followed by an underscore.
+
+```{code-cell}
+# Find all column names from the PHZ table.
+phz_columns = [name for name in schema.names if name.startswith("PHZ_")]
+print(f"{len(phz_columns)} columns from the PHZ table. First four are:")
+phz_columns[:4]
+```
+
+Column metadata includes unit and description.
+
+```{code-cell}
+schema.field("MER_FLUX_Y_2FWHM_APER").metadata
+```
+
+Euclid Q1 offers many flux measurements, both from Euclid detections and from external ground-based surveys.
+They are given in microjanskys, so all flux columns can be found by searching the metadata for this unit.
+
+```{code-cell}
+# Find all flux columns.
+flux_columns = [field.name for field in schema if field.metadata[b"unit"] == b"uJy"]
+print(f"{len(flux_columns)} flux columns. First four are:")
+flux_columns[:4]
+```
+
+Columns associated with external surveys are identified by the inclusion of "EXT" in the name.
+
+```{code-cell}
+external_flux_columns = [name for name in flux_columns if "EXT" in name]
+print(f"{len(external_flux_columns)} flux columns from external surveys. First four are:")
+external_flux_columns[:4]
+```
+
+### A.2 Additional columns
+
++++
+
+In addition to the columns from Euclid Q1 tables, the following columns have been added to this dataset:
+
+- 'TILEID' : Euclid MER tile index.
+- '_healpix_9' : HEALPix order 9 pixel index. Useful for spatial queries.
+- '_healpix_19' : HEALPix order 19 pixel index. Useful for spatial queries.
+- '_healpix_29' : (hats column) HEALPix order 29 pixel index. Useful for spatial queries.
+- 'Norder' : (hats column) HEALPix order at which the data is partitioned.
+- 'Npix' : (hats column) HEALPix pixel index at order Norder.
+- 'Dir' : (hats column) Integer equal to 10_000 * floor[Npix / 10_000].
+
```

Comment:
[Optional] Here you can emphasize on the last 3 partitioning columns by doing `dataset.partitioning.schema` in a cell

Reply:
Good idea, thanks.